### PR TITLE
light: remove global context from tests

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: light/client_test.go [hunk 1]
```diff
@@ -206,9 +205,15 @@ func TestClient_SequentialVerification(t *testing.T) {
 		},
 	}
 
+	bctx, bcancel := context.WithCancel(context.Background())
```
Comment (creachadair) [inline light/client_test.go:208]: Could we do this inside the test case? I don't think we need a "root" context here (unless I missed something).

File: light/client_test.go [hunk 2]
```diff
@@ -1003,9 +1059,14 @@ func TestClientEnsureValidHeadersAndValSets(t *testing.T) {
...
```
Comment (creachadair) [inline light/client_test.go:1062]: Here again, I think maybe we don't need the root context.

**Review by creachadair — COMMENTED** [review]
Yay, context. There are a few places where I think we could simplify, but on balance LGTM.

File: light/client_test.go (hunk 1 above)
Comment (tychoish) [reply light/client_test.go:208]: yeah, I just did this by default

**Review by creachadair — APPROVED** [review]

**Pull request merged**